zendesk/sideload/sideload.go

```diff
@@ -0,0 +1,69 @@
+// sideload Allows for sideload support when calling the zendesk api.
+// For more information about sideloading see: https://developer.zendesk.com/rest_api/docs/support/side_loading
+package sideload
+
+import (
+	"fmt"
+	"github.com/iancoleman/strcase"
+	"reflect"
+)
+
+type SideLoader interface {
+	Key() string
+}
+
+type ExtraObjectSideloader interface {
+	SideLoader
+	AppendToStruct([]reflect.StructField) []reflect.StructField
+	IsAssignable() bool
+	SetValue(reflect.Value)
+}
+
+type simpleInclude string
+
+func (s simpleInclude) Key() string {
+	return string(s)
+}
+
+// In some instances sideloading results in extra fields being added to the main object returned by the api
+// The Include sideloader should be used in such cases
+func Include(key string) SideLoader  {
```

Comment:
An example API response can be found here https://gist.github.com/tamccall/13f98c84a6175ac333d3af83b3689ed5

In this example if the ticket api is told to include `dates` the ticket object itself is modified to have an additional dates property. The way i could see supported this would be:

-  to just include the Dates object on the ticket struct
- Allow the client to provide a JsonPath that tells us where to find the sideloaded object in the body and allow them to provide an object reference for us to marshal it into.

Wondering what we think is a better approach.

Reply:
I would go with either way tbh. However, I think the second approach makes more sense as it provides more flexibility to keep the sideloaded data decoupled from the main ticket object.

Reply:
@abebars The second approach was more or less what I was trying to go for with #154. maybe the best thing to do at this point would be to close this pr and just leave that one open.